src/Components/GrowthChart/GrowthChart.tsx

```diff
@@ -1,25 +1,53 @@
-import React, { useMemo, useState } from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { GrowthChartBuilder } from './GrowthChartBuilder';
-import { chartData } from '../../DataSets/WhoStandardDataSets/ZScores/ChartDataZscores';
-import { useRangeTimePeriod } from './useRangeTimePeriod';
+import { useRangeTimePeriod } from '../../utils/useRangeTimePeriod';
 import { ChartSelector } from '../GrowthChartSelector';
-import { CategoryCodes, ChartData } from '../../types/chartDataTypes';
+import { ChartData, GenderCodes, CategoryCodes } from '../../types/chartDataTypes';
 import { useCalculateMinMaxValues } from '../../utils/useCalculateMinMaxValues';
 import { GrowthChartAnnotations } from './GrowthChartOptions';
 import { ChartSettingsButton } from './ChartSettingsButton';
+import { useChartDataForGender } from '../../utils/DataFetching/Sorting/useChartDataForGender';
+import { useTeiById } from '../../utils/DataFetching/Hooks';
 
-export const GrowthChart = () => {
-    const [category, setCategory] = useState<keyof typeof CategoryCodes>(Object.keys(chartData)[0] as keyof typeof CategoryCodes);
-    const [dataset, setDataset] = useState<keyof ChartData>(Object.keys(chartData[category].datasets)[0] as keyof ChartData);
+interface GrowthChartProps {
+    teiId: string;
+}
 
-    const categoryDataSets = chartData[category];
-    const dataSetEntry = categoryDataSets.datasets[dataset];
+export const GrowthChart = ({ teiId }: GrowthChartProps) => {
+    const {
+        trackedEntity,
+        isLoading,
+        isError,
+    } = useTeiById({ teiId });
+    const trackedEntityGender = GenderCodes[trackedEntity?.attributes.find(
+        (attribute: any) => attribute.displayName === 'Gender',
+    ).value?.toLowerCase() as 'male' | 'female'];
```

Comment:
This will hopefully improve with the implementation of a config file containing the ID for the Gender value, so I wont have to find an attribute with "Gender" as displayname..